airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -511,21 +514,47 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
             ),
         ]
 
-        return [
-            StreamFacade.create_from_stream(stream, self, entrypoint_logger, self._create_empty_state(), NoopCursor())
-            if stream.name in self._streams_configured_as_full_refresh
-            else stream
-            for stream in streams
-        ]
+        state_manager = ConnectorStateManager(stream_instance_map={s.name: s for s in streams}, state=self._state)
+        return [self._to_concurrent(stream, self._start_date_to_timestamp(config), state_manager) for stream in streams]
+
+    def _to_concurrent(self, stream: Stream, fallback_start, state_manager: ConnectorStateManager) -> Stream:
+        if os.environ.get("SKIP_CONCURRENCY"):
+            return stream
+        if stream.name in self._streams_configured_as_full_refresh:
+            return StreamFacade.create_from_stream(stream, self, entrypoint_logger, self._create_empty_state(), NoopCursor())
+
+        state = state_manager.get_stream_state(stream.name, stream.namespace)
+        slice_boundary_fields = self._SLICE_BOUNDARY_FIELDS_BY_IMPLEMENTATION.get(type(stream))
+        if slice_boundary_fields:
+            cursor_field = CursorField(stream.cursor_field) if isinstance(stream.cursor_field, str) else CursorField(stream.cursor_field[0])
+            converter = EpochValueConcurrentStreamStateConverter()
+            cursor = ConcurrentCursor(
+                stream.name,
+                stream.namespace,
+                state_manager.get_stream_state(stream.name, stream.namespace),
+                self.message_repository,
+                state_manager,
+                converter,
+                cursor_field,
+                slice_boundary_fields,
+                fallback_start,
+            )
+            return StreamFacade.create_from_stream(stream, self, entrypoint_logger, state, cursor)
+
+        return stream
 
     def _create_empty_state(self) -> MutableMapping[str, Any]:
         # The state is known to be empty because concurrent CDK is currently only used for full refresh
         return {}
 
     @staticmethod
-    def _start_date_to_timestamp(start_date: str) -> int:
+    def _start_date_to_timestamp(config: Mapping[str, Any]) -> int:
+        if "start_date" not in config:
+            return pendulum.datetime(2017, 1, 25).int_timestamp  # type: ignore  # pendulum not typed
```

Comment:
I know it predated your code, but any idea where the time comes from? If so, mind adding a comment?

Reply:
Unfortunately, I have no clue and dates like that seems dangerous to me as it makes the initial syncs longer and longer as time goes by